artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/proton/AMQPLargeMessageReader.java

```diff
@@ -34,7 +34,7 @@ public class AMQPLargeMessageReader implements MessageReader {
 
    private final ProtonAbstractReceiver serverReceiver;
 
-   private AMQPLargeMessage currentMessage;
+   private volatile AMQPLargeMessage currentMessage;
```

Comment:
@tabish121 , @gemmellr currentMessage is set to null as part of the Session Executor. 
We shouldn't have multiple threads messing with the variable at the same time as we set readable to false and transfer the execution, but I think it would be safer to set it as volatile.


Let me know if you have a concern here.

Reply:
I'd go with abundance of caution since we are crossing threads where we weren't before.